Pull Requests Size Guidance (#160)
microsoft/code-with-engineering-playbook: code-reviews/readme.md

2. In the daily standup meeting check tasks pending for review and make sure they have reviewers assigned.
3. Junior teams and teams new to the process can consider creating separate tasks for reviews together with the tasks themselves.

### Size guidance

> We should always aim to have Pull Requests as smaller as possible, without losing context and technical feasibility. Small pull requests have multiple benefits:

Reviewer: ```suggestion
We should always aim to have Pull Requests as small as possible, without losing context and technical feasibility. Small pull requests have multiple advantages:
```

> We should always aim to have Pull Requests as smaller as possible, without losing context and technical feasibility. Having small Pull Requests, have multiple benefits:

Reviewer 2: Having small Pull Requests, have multiple benefits: -> Small pull requests have multiple benefits:

Author: Fixed.

> 1. It's easier to review, a clear benefit for the reviewers.

Reviewer: ```suggestion
1. It's easier to review; a clear benefit for the reviewers.
```

2. It's easier to deploy, at least in terms of a bigger Pull Request. This is aligned with the strategy of release fast and release often.

> 3. Minimizes possible conflicts, or stale Pull Requests difficult to merge and keep in sync with master either because it's very dynamic or because it has some refactoring.

Reviewer: ```suggestion
3. Minimizes possible conflicts and stale Pull Requests, which are difficult to merge and keep in sync with master either because they're very dynamic or contain refactoring.
```

Reviewer 3: NIT: "Minimize possible conflicts..."

> However, we should be aware to avoid having Pull Requests that include code without context or very loose coupled. 

Reviewer: ```suggestion
However, we should avoid having Pull Requests that include code that is without context or loosely coupled. 
```

> There are times where seem a big Pull Request is unavoidable, however, there are some strategies to keep Pull Requests small depending on the "cause" of the ineluctability:

Reviewer: ```suggestion
There are times where seem a big Pull Request is unavoidable; however, there are some strategies to keep Pull Requests small depending on the "cause" of the ineluctability:
```

Reviewer 3: NIT: "There are times when large PRs seem unavoidable. However, there are strategies to keep PRs small...."

> #### Minimum Working Components

Reviewer 3: Seems a little funny to be mentioning these strategies to keep PRs small without expanding on them. I suggest adding a little information beneath these sections.
